src/target/riscv/program.c

```diff
@@ -67,11 +70,18 @@ int riscv_program_exec(struct riscv_program *p, struct target *t)
 	if (riscv_program_write(p) != ERROR_OK)
 		return ERROR_FAIL;
 
-	if (riscv_execute_debug_buffer(t) != ERROR_OK) {
+	unsigned int abstractcs_err;
+	if (riscv_execute_debug_buffer(t, &abstractcs_err) != ERROR_OK) {
+		p->execution_result = (abstractcs_err == DM_ABSTRACTCS_CMDERR_EXCEPTION)
+			? RISCV_DBGBUF_EXEC_RESULT_EXCEPTION
+			: RISCV_DBGBUF_EXEC_RESULT_UNKNOWN;
```

Comment:
UNKNOWN doesn't feel right here. We don't know what happened, but it was definitely an error. UNKNOWN_ERROR seems more descriptive.